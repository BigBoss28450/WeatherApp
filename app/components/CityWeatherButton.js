import React from "react";
import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
import colors from "../config/colors";
import { kelvinToCelsius } from "../config/helpers";

export default function CityWeatherButton({ city, temp, onPress }) {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.weatherButtonContainer}>
        <Text style={{ color: colors.white }}>{city}</Text>
        <Text style={{ color: colors.white }}>{kelvinToCelsius(temp)} CÂº</Text>
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  weatherButtonContainer: {
    borderWidth: 1,
    borderColor: colors.white,
    borderRadius: 5,
    padding: 20,
    marginHorizontal: 20,
    marginBottom: 20,
    flexDirection: "row",
    justifyContent: "space-between",
  },
});
