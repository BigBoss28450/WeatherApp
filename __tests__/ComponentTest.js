import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import CityWeatherButton from "../app/components/CityWeatherButton";

jest.useFakeTimers();

test("Link changes the class when hovered", () => {
  const component = renderer.create(
    <CityWeatherButton city="Colima" temp={25} onPress={() => true} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.name = "Colima";
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.temp = 25;
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
