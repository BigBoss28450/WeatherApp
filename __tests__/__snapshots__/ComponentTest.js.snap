// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#FFFFFF",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 20,
        "marginHorizontal": 20,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
    >
      Colima
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
    >
      -248
       Cº
    </Text>
  </View>
</View>
`;

exports[`Link changes the class when hovered 2`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#FFFFFF",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 20,
        "marginHorizontal": 20,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
    >
      Colima
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
    >
      -248
       Cº
    </Text>
  </View>
</View>
`;

exports[`Link changes the class when hovered 3`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#FFFFFF",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 20,
        "marginHorizontal": 20,
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
    >
      Colima
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
    >
      -248
       Cº
    </Text>
  </View>
</View>
`;
